src/app/hooks/usePupils.ts

```diff
@@ -19,23 +23,56 @@ export default function usePupils(): {
     return pupils.find((pupil) => pupil.name === name);
   }
 
-  function addPupil(pupil: { name: string; evaluation: string }) {
+  function addPupil(pupil: {
+    name: string;
+    category: string;
+    evaluation: string;
+  }) {
     const existingPupil = findPupilByName(pupil.name);
     if (existingPupil) {
       const existingPupilId = pupils.findIndex(
         (pupil) => pupil.name === existingPupil.name
       );
-      existingPupil.evaluations = [
-        ...existingPupil.evaluations,
-        {
-          id: nanoid(),
```

Comment:
I just looked up the difference between nanoID & UUID. Great choice, though!

Reply:
It was rather the idea of Marwin, but maybe this is another good reason :D